andela/knights-bn-backend
# #170947663 User will be able to view and edit profile information

#### What does this PR do?
- Adding functionality of allowing user to view his/her profile information and edit them.
#### Description of Task to be completed?
- A user will have a profile page that will display  information like `names`, `email`, `gender`, `passport number`, `department`, `preferred currency`, `preferred language`, `user role`, `where you live`, `birthday`, `line manager`,` biography` and upload `profile image`. He/she will be able to view his/her own information and be able to edit them.
#### How should this be manually tested?
- open your terminal and run this command `git clone https://github.com/andela/knightsbn    backend.git` to clone repository
- run `cd knights-bn-backend` to get inside the project
- run `git fetch && git checkout ft-profile-page-settings-170947663` 
- run `npm install` to install dependencies
- check if you have PostgresSQL database if you don't have it install it from here [PostgreSQL](https://www.2ndquadrant.com/en/blog/pginstaller-install-postgresql/)
- create database
- create `.env` file in your project root directory
- follow `env.example` file to create environment variables in `.env` file and 
   assign them with all your database configurations
- run `npm run createTables` to create tables in your new database you have created.
- run `npm run test` to run all the tests
- run `npm run start` to start the server.
- use swagger documentation to test every end point by visiting this link `http://localhost:4000/api/v1/docs` on your browser or follow the next steps
- open your postman or any other API tester you have
-  to view or edit user profile you must be logged in or signed up , see how to do [log in](https://github.com/andela/knights-bn-backend/pull/23) or [sign up](https://github.com/andela/knights-bn-backend/pull/17)
- to test view profile information send GET request to this link `http://localhost:4000/api/v1/user/profile`
- to test update profile information send PATCH request to this link `http://localhost:4000/api/v1/edit/user/profile` and use form-data option under the body in your postman in order to find a way you can upload a profile image  and click on `Bulk Edit` option on the right part under save button and past the information bellow to generate a well define form,  after pasting the data click on `Key-Value Edit` to generate a form,  and fill the fields and remember to change profileImage field type to file so that you can upload image. after filling every field send your request.
```
firstName:eugene
lastName:rutikanga
email:eukigali@gmail.com
passport:12348934
gender:male
birthDay:1990-2-3
language:English
currency:Euro
homeTown:dallas
department:IT
lineManager:Peter Doe
biography:Lorem Ipsum is simply dummy text of the printing and typesetting industry
```

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#170947663](https://www.pivotaltracker.com/story/show/170947663)

#### Screenshots
get profile information

<img width="959" alt="Screen Shot 2020-02-24 at 22 30 46" src="https://user-images.githubusercontent.com/36619897/75188804-84331b80-5755-11ea-976d-85267f13e010.png">


update the information

<img width="1156" alt="Screen Shot 2020-02-24 at 22 24 25" src="https://user-images.githubusercontent.com/36619897/75188265-8052c980-5754-11ea-98d2-2a5d6df212ea.png">
